up-transport-vsomeip/src/transport.rs

```diff
@@ -0,0 +1,566 @@
+                    .validate(attributes)
+                    .map_err(|e| {
+                        let msg = format!("Wrong Response UAttributes: {e:?}");
+                        log::error!("{msg}");
+                        UStatus::fail_with_code(UCode::INVALID_ARGUMENT, msg)
+                    })?;
+            }
+            UMessageType::UMESSAGE_TYPE_UNSPECIFIED => {
+                let msg = "Wrong Message type in UAttributes".to_string();
+                log::error!("{msg}");
+                return Err(UStatus::fail_with_code(UCode::INVALID_ARGUMENT, msg));
+            }
+        }
+
+        trace!("Sending message: {:?}", message);
+
+        let Some(source_filter) = message.attributes.source.as_ref() else {
+            return Err(UStatus::fail_with_code(
+                UCode::INVALID_ARGUMENT,
+                "UMessage provided with no source",
+            ));
+        };
+
+        source_filter.verify_no_wildcards().map_err(|e| {
+            UStatus::fail_with_code(UCode::INVALID_ARGUMENT, format!("Invalid source: {e:?}"))
+        })?;
+
+        let sink_filter = message.attributes.sink.as_ref();
+
+        if let Some(sink) = sink_filter {
+            sink.verify_no_wildcards().map_err(|e| {
+                UStatus::fail_with_code(UCode::INVALID_ARGUMENT, format!("Invalid sink: {e:?}"))
+            })?;
+        }
+
+        let message_type = determine_send_type(source_filter, &sink_filter.cloned())?;
+        trace!("inside send(), message_type: {message_type:?}");
+        let app_name = find_app_name(message_type.client_id()).await;
+        trace!("app_name: {app_name:?}");
+
+        self.initialize_vsomeip_app_as_needed(&message_type, app_name)
+            .await?;
+
+        let app_name = format!("{}", message_type.client_id());
+        trace!("app_name: {app_name}");
+
+        self.register_for_returning_response_if_point_to_point_listener_and_sending_request(
+            &message,
+            source_filter,
+            sink_filter,
+            message_type.clone(),
+        )
+        .await?;
+
+        let (tx, rx) = oneshot::channel();
+        send_to_inner_with_status(
+            &self.inner_transport.transport_command_sender,
+            TransportCommand::Send(message, message_type, tx),
+        )
+        .await?;
+        await_internal_function(UP_CLIENT_VSOMEIP_FN_TAG_SEND_INTERNAL, rx).await
+    }
+
+    async fn register_listener(
+        &self,
+        source_filter: &UUri,
+        sink_filter: Option<&UUri>,
+        listener: Arc<dyn UListener>,
+    ) -> Result<(), UStatus> {
+        let registration_type_res =
+            determine_registration_type(source_filter, &sink_filter.cloned(), self.ue_id);
+        let Ok(registration_type) = registration_type_res else {
+            return Err(UStatus::fail_with_code(UCode::INVALID_ARGUMENT, "Invalid source and sink filters for registerable types: Publish, Request, Response, AllPointToPoint"));
+        };
```

Comment:
```suggestion
        let registration_type =
            determine_registration_type(source_filter, &sink_filter.cloned(), self.ue_id)
            .map_err(|e| UStatus::fail_with_code(UCode::INVALID_ARGUMENT, "Invalid source and sink filters for registerable types: Publish, Request, Response, AllPointToPoint"))?;
```

Reply:
Resolved the error handling within determine_message_type a bit more effectively, so this issue goes away.